app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
+import { browserHistory } from 'react-router'
+import PolicyPicker from '../../shared/PolicyPicker/PolicyPicker.jsx'
+import VaultObjectDeleter from '../../shared/DeleteObject/DeleteObject.jsx'
+
+function snackBarMessage(message) {
+    let ev = new CustomEvent("snackbar", { detail: { message: message } });
+    document.dispatchEvent(ev);
+}
 
 export default class GithubAuthBackend extends React.Component {
+    backendConfigSchema = {
+        organization: '',
+        base_url: undefined,
+        max_ttl: undefined,
+        ttl: undefined
+    }
+
+    itemConfigSchema = {
+        id: '',
+        policies: []
+    }
+
     constructor(props) {
         super(props);
         this.state = {
-            requestOrganization: false,
-            organization: window.localStorage.getItem('githubOrganization') || '',
-            teamName: '',
-            policies: [],
-            checkedPolicies: [],
-            submitBtnColor: 'lightgrey',
-            submitBtnDisabled: true,
-            errorMessage: '',
-            selected: props.selected === 'Github'
+            baseUrl: `/auth/github/${this.props.params.namespace}/`,
+            baseVaultPath: `auth/${this.props.params.namespace}`,
+            teams: [],
+            users: [],
+            config: this.backendConfigSchema,
+            newConfig: this.backendConfigSchema,
+            itemConfig: this.teamConfigSchema,
+            selectedItemId: '',
+            newItemId: '',
+            isBackendConfigured: false,
+            openItemDialog: false,
+            selectedTab: "team",
+            deleteUserPath: ''
         };
 
-        this.processTeamNameDebounced = _.debounce(this.processTeamName, 400);
-
         _.bindAll(
             this,
-            'listPolicies',
-            'renderPolicies',
-            'policyChecked',
-            'teamNameChanged',
-            'submitGithubPolicy',
-            'requestOrganization',
-            'renderOrganizationDialog',
-            'getTeamPolicy',
-            'processTeamName'
+            'listGithubTeams',
+            'listGithubUsers',
+            'getOrgConfig',
+            'displayItem'
         );
-
     }
 
-    componentWillMount() {
-        this.requestOrganization();
-        this.listPolicies();
+    listGithubTeams() {
+        tokenHasCapabilities(['list'], `${this.state.baseVaultPath}/map/teams`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/map/teams`, { list: true }, null)
+                    .then((resp) => {
+                        let teams = resp.data.data.keys;
+                        this.setState({ teams: _.valuesIn(teams) });
+                    })
+                    .catch((error) => {
+                        if (error.response.status !== 404) {
+                            snackBarMessage(error);
+                        } else {
+                            this.setState({ teams: [] });
```

Comment:
Then it will display the error out in the snack bar, 404 is a specific exception which in this case means the config hasn't been initialized